plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts

```diff
@@ -26,18 +27,40 @@ import {
 } from './types';
 import { runDockerContainer, runCommand } from './helpers';
 
-const commandExists = require('command-exists-promise');
+type TechdocsGeneratorOptions = {
+  // This option enables users to configure if they want to use TechDocs container
+  // or generate without the container.
+  // This is used to avoid running into Docker in Docker environment.
+  useTechdocsContainer: boolean;
```

Comment:
Could this be driven from app-config.yaml instead?

Reply:
It could, but i thought it made sense to pass it in to the generator when you set it up in the backends techdocs file. Do you think the app-config is better?

Reply:
I think it will be better -- case 1: in local machine you are using docker when you do development for example. and in production app-config you can override it with local setup instead of docker.

Reply:
I just like to keep all config in one place in general. I'll let you decide :) When people ask how to make use of a local runner it's kind of nice to just say to change a flag.

Reply:
The reason why i thought this would be a better way of doing it is that it kinda feels like bloating the app-config. It makes sense though. Will update PR. :)

Reply:
Another thought: is it even time to make a field `generator: "local"` instead of a specific bool? :) for ze fuuutuuuuure with more potential runners. But yeah yeah, bikeshedding at this point :)

Reply:
I'm good eater way. I can see advantages with both solutions. If it's correct that you can just assign the bool, by using something like `process.env.MY_BOOL` in the backend techdocs file, then I think we should keep it away from the app-config, to not bloat it. Because then you can set a env variable to different things in dev and prod as @pavel-kurnosov requested.

Also - this is just a thought - maybe it would be relevant in the feature to make it more modular like the preparers. So you could register different generators depending on your needs. Both in regards to different methods for generating mkdocs (e.g. local vs docker container), but also maybe for generating different docs types, if you want to use something other than mkdocs for documentation. But that's probably a whole other discussion 🙂

Reply:
I think setting the value to string instead with local/docker is a good approach -- current setup is still not the best both local and docker are not the best options. In the future, I see at least remote and event as other options -- where I can install container separately or subscribe to an event and do whatever I want :)